Near completion generated image disappears[Bug]:

### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

5 days ago an update occured which totally broke my version of Stable Diffusion. As the image is nearly completed generating it disappears and this Error message appears: RuntimeError: Expected query, key, and value to have the same dtype, but got query.dtype: float key.dtype: float and value.dtype: struct c10::Half instead.
Time taken: 8.31sTorch active/reserved: 2226/2274 MiB, Sys VRAM: 3929/11264 MiB (34.88%)
Uninstalled then re-installed Git, Python, the webui - Nothing helps. At this point I would be happy if someone just guided me to wipe out everything and re-install a working version of Automatic1111.

### Steps to reproduce the problem

1. Go to .... 
2. Press ....
3. ...
NA

### What should have happened?

NA

### Commit where the problem happens

NA

### What Python version are you running on ?

Python 3.10.x

### What platforms do you use to access the UI ?

Windows

### What device are you running WebUI on?

Nvidia GPUs (GTX 16 below)

### What browsers do you use to access the UI ?

Microsoft Edge

### Command Line Arguments

```Shell
No
```


### List of extensions

No

### Console logs

```Shell
venv "G:\stable-diffusion-webui-master\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.3.0
Commit hash: 20ae71faa8ef035c31aa3a410b707d792c8203a3
Installing requirements
Launching Web UI with arguments: --medvram --autolaunch
No module 'xformers'. Proceeding without it.
reading lora G:\stable-diffusion-webui-master\models\Lora\None.safetensors: AssertionError
Traceback (most recent call last):
  File "G:\stable-diffusion-webui-master\extensions-builtin\Lora\lora.py", line 83, in __init__
    self.metadata = sd_models.read_metadata_from_safetensors(filename)
  File "G:\stable-diffusion-webui-master\modules\sd_models.py", line 230, in read_metadata_from_safetensors
    assert metadata_len > 2 and json_start in (b'{"', b"{'"), f"{filename} is not a safetensors file"
AssertionError: G:\stable-diffusion-webui-master\models\Lora\None.safetensors is not a safetensors file

Loading weights [fba9027fa4] from G:\stable-diffusion-webui-master\models\Stable-diffusion\realisian_v40.safetensors
Creating model from config: G:\stable-diffusion-webui-master\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
DiffusionWrapper has 859.52 M params.
Startup time: 6.5s (import torch: 1.8s, import gradio: 1.1s, import ldm: 0.8s, other imports: 0.9s, load scripts: 0.9s, create ui: 0.5s, gradio launch: 0.3s).
Applying optimization: sdp-no-mem... done.
Textual inversion embeddings loaded(64): AlexandraDaddario, AliciaSilverstone, AlisonBrie, AllieDunn, AmberHeard, AnaDeArmas, AnastasiaIvanova, AnnaKendrick, ArianaGrande, AshleyResch, BarbaraPalvin, BrineySpears, BrookeShields, CameronDiaz, CamilaMorrone, CharlizeTheron_ti, CourtneyCox, DaisyRidley, DaisyRidley2, EmiliaClarke, EmiliaClarke2, EmiliaClarke3, EmmaWatson, FrankFrazetta, GeenaDavis, GemmaArterton, GoerginaMazzeo, HayleyAtwell, HenryCavill, HuangGuangJian, JaniceGriffith, JennaOrtega, JenniferConnelly, JenniferLawrence, JessicaAlba, KateBeckinsale, KeanuReeves, KristenBell, KristenStewart, Kuvshinov, LittleCaprice, LivTyler, LivvyDunne, LucyHale, LucyLiu, LynCarver, MeganFox, MelissaBenoist, MichelleYeoh, MilaKunis, MilaMilenka, MillaJovovich, MollyQuinn, MorenaBaccarin, NataLee, NataliaDyer, NataliePortman, RebeccaFerguson, SadieSink, ScarlettJohansson, ShaniaTwain, SummerGlau, VeronicaMars, YvonneStrahovski
Textual inversion embeddings skipped(1): AnaDeArmas2
Model loaded in 2.8s (load weights from disk: 0.7s, create model: 0.5s, apply weights to model: 0.6s, apply half(): 0.7s, load textual inversion embeddings: 0.4s).
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:08<00:00,  2.33it/s]
Error completing request███████████████████████████████████████████████████████████████| 20/20 [00:06<00:00,  2.81it/s]
Arguments: ('task(fbbqb7n0ofub32z)', 'a cat', '', [], 20, 0, False, False, 1, 1, 7, -1.0, -1.0, 0, 0, 0, False, 512, 512, False, 0.7, 2, 'Latent', 0, 0, 0, 0, '', '', [], 0, False, False, 'positive', 'comma', 0, False, False, '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, 0) {}
Traceback (most recent call last):
  File "G:\stable-diffusion-webui-master\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "G:\stable-diffusion-webui-master\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "G:\stable-diffusion-webui-master\modules\txt2img.py", line 57, in txt2img
    processed = processing.process_images(p)
  File "G:\stable-diffusion-webui-master\modules\processing.py", line 611, in process_images
    res = process_images_inner(p)
  File "G:\stable-diffusion-webui-master\modules\processing.py", line 731, in process_images_inner
    x_samples_ddim = [decode_first_stage(p.sd_model, samples_ddim[i:i+1].to(dtype=devices.dtype_vae))[0].cpu() for i in range(samples_ddim.size(0))]
  File "G:\stable-diffusion-webui-master\modules\processing.py", line 731, in <listcomp>
    x_samples_ddim = [decode_first_stage(p.sd_model, samples_ddim[i:i+1].to(dtype=devices.dtype_vae))[0].cpu() for i in range(samples_ddim.size(0))]
  File "G:\stable-diffusion-webui-master\modules\processing.py", line 519, in decode_first_stage
    x = model.decode_first_stage(x)
  File "G:\stable-diffusion-webui-master\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "G:\stable-diffusion-webui-master\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "G:\stable-diffusion-webui-master\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "G:\stable-diffusion-webui-master\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 826, in decode_first_stage
    return self.first_stage_model.decode(z)
  File "G:\stable-diffusion-webui-master\modules\lowvram.py", line 52, in first_stage_model_decode_wrap
    return first_stage_model_decode(z)
  File "G:\stable-diffusion-webui-master\repositories\stable-diffusion-stability-ai\ldm\models\autoencoder.py", line 90, in decode
    dec = self.decoder(z)
  File "G:\stable-diffusion-webui-master\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "G:\stable-diffusion-webui-master\repositories\stable-diffusion-stability-ai\ldm\modules\diffusionmodules\model.py", line 631, in forward
    h = self.mid.attn_1(h)
  File "G:\stable-diffusion-webui-master\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "G:\stable-diffusion-webui-master\modules\sd_hijack_optimizations.py", line 620, in sdp_no_mem_attnblock_forward
    return sdp_attnblock_forward(self, x)
  File "G:\stable-diffusion-webui-master\modules\sd_hijack_optimizations.py", line 612, in sdp_attnblock_forward
    out = torch.nn.functional.scaled_dot_product_attention(q, k, v, dropout_p=0.0, is_causal=False)
RuntimeError: Expected query, key, and value to have the same dtype, but got query.dtype: float key.dtype: float and value.dtype: struct c10::Half instead.
```


### Additional information

_No response_

--- a/modules/sd_hijack_optimizations.py
+++ b/modules/sd_hijack_optimizations.py
@@ -605,7 +605,7 @@ def sdp_attnblock_forward(self, x):
     q, k, v = (rearrange(t, 'b c h w -> b (h w) c') for t in (q, k, v))
     dtype = q.dtype
     if shared.opts.upcast_attn:
-        q, k = q.float(), k.float()
+        q, k, v = q.float(), k.float(), v.float()
     q = q.contiguous()
     k = k.contiguous()
     v = v.contiguous()

